src/librustc/session/config.rs

```diff
@@ -1901,6 +1913,21 @@ pub fn build_session_options_and_crate_config(
     let crate_types = parse_crate_types_from_list(unparsed_crate_types)
         .unwrap_or_else(|e| early_error(error_format, &e[..]));
 
+    if matches.opt_present("extern-public") && !nightly_options::is_nightly_build() {
```

Comment:
I think this should probably be gated on requiring `-Z unstable-options`, as that's typically how we've been gating unstable CLI flags historically. Passing `-Z unstable-options` will itself require transitively the `is_nightly_build()` option